src/metabase/api/user.clj

```diff
@@ -80,7 +73,8 @@
   ;; only allow updates if the specified account is active
   (api/check-404 (db/exists? User, :id id, :is_active true))
   ;; can't change email if it's already taken BY ANOTHER ACCOUNT
-  (api/check-400 (not (db/exists? User, :email email, :id [:not= id])))
+  (api/check (not (db/exists? User, :email email, :id [:not= id]))
+    [400 "Email address already associated to another user."])
```

Comment:
This should conform to our errors conventions, either `{ errors: { email: "Email address ..." } }` if it's relevant to a specific field or `{ message: "Email address ..." }` if it applies to the whole request (the former in this case)

Reply:
Copied from conversation with @camsaul:

> Writing all the stuff out is annoying so stick a helper function in `metabase.api.common` util like

```
(defn check-field [condition field-kw error-message]
  (when condition
    (throw (ex-info error-message {:status-code 400, :errors {field-kw error-message}}))))
```

> then you could do

```
(check-field (not (db/exists? User :email email)) :email (tru "Email address already in use."))
```

Reply:
omg @tlrobinson I'm a moron I forgot we already have a function that does exactly that. It's called `api/checkp`. It takes the exact same params